packages/flutter_tools/test/commands/packages_test.dart

```diff
@@ -145,32 +147,35 @@ class MockProcessManager implements ProcessManager {
     bool runInShell: false,
     ProcessStartMode mode: ProcessStartMode.NORMAL,
   }) {
+    commands = command;
     return new Future<Process>.value(processFactory(command));
   }
 
   @override
   dynamic noSuchMethod(Invocation invocation) => null;
 }
 
-/// Process that prompts the user to proceed, then asynchronously writes
+/// A process that prompts the user to proceed, then asynchronously writes
 /// some lines to stdout before it exits.
 class PromptingProcess implements Process {
-  PromptingProcess(String prompt, List<String> outputLines) {
-    _stdoutController.add(UTF8.encode(prompt));
+  Future<Null> showPrompt(String prompt, List<String> outputLines) async {
     _stdin.future.then((List<int> bytes) async {
       // echo stdin to stdout
```

Comment:
I think it would read better as:
```
_stdoutController.add(UTF8.encode(prompt));
var bytes = await _stdin.future;
if (bytes[0] == UTF8.encode('y')[0]) {
  _stdoutController.add(UTF8.encode('$line\n'));
}
await _stdoutController.close();
```